### ISSUE TITLE
[BUG] Home view is always "offline" but everything else works

### ISSUE THREAD

**Issue by carlosefr** [mastodon/mastodon-ios#898]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

For at least the last 12 hours or so, I've been getting a continuous "offline" indicator (as pictured). But only for the home view, as I can see notifications, my own profile, favorites, search, and even post just fine. I have a mastodon.social account and the web site seems fine (also installed Metatext to check, and it also works).

I've tried logging off in the app, and even deleting it and reinstalling, to no avail.

This looks like a bug, but I don't know how to further debug this (the app provides no clue as why it may be "offline").

![238fa6abedcd88cf](https://user-images.githubusercontent.com/937276/213693692-a060f5de-6f7c-4874-aab2-0b2c642915b0.png)

### Environment

```markdown
- Device: iPhone SE
- OS: iOS 15.7.2
- Version: 1.4.9
- Build: 224
```

**Comment by smjklake** [comment]
I am currently having the same issue on iPhone 12 Pro iOS 16.2. I did not have Metatext installed so I installed it to see if I had a similar problem there and with a fresh installation of Metatext I am receiving an error "The Data couldn't be read becuase it is missing" when trying to load data from the home feed. Local and Federated timelines still work as expected. At this moment it is looking like it may be an issue with my particular instance or mastodon in general more than the client.

**Comment by j-f1** [comment]
Which servers are you using? (and/or what version of Mastodon are they using?) They might not support the API calls that the app is using. That should probably be fixed by the app though!

**Comment by smjklake** [comment]
I'm on fosstodon.org Mastodon v4.0.2. I also tried Ice Cubes just now and that one seems to work with no issues.

**Comment by haydn** [comment]
I'm seeing this too.

- Mastodon for iOS v1.4.9
- iOS v16.1.1
- iPhone 8
- Server: indieweb.social

**Comment by LeBenLeBen** [comment]
I had the same issue yesterday and this morning, after a while I pull to refresh and it worked. While browsing, it came back to "offline" for a while and now I was able to refresh again. I would think something odd happens with the instance but we all seems to be on a different one.

I’m on mastodon.social (4.0.2) with Mastodon 1.4.9 on an iPhone X with iOS 16.2

**Comment by papproth** [comment]
Same over here, permanent offline badge at the top of the screen. I haven‘t seen this ever before.

- Mastodon for iOS v1.4.9
- iOS v16.2
- iPhone SE 2nd
- Server: mastodon.social, v4.0.2

Any ideas on how to gather more insights into what is going wrong?

**Comment by b00ga** [comment]
+1

* App v1.4.9
* iOS 16.1.1
* iPhone 8 Plus
* infosec.exchange

**Comment by papproth** [comment]
> Same over here, permanent offline badge at the top of the screen. I haven‘t seen this ever before.
> 
> * Mastodon for iOS v1.4.9
> * iOS v16.2
> * iPhone SE 2nd
> * Server: mastodon.social, v4.0.2
> 
> Any ideas on how to gather more insights into what is going wrong?

Issue seems gone tonight.
Still I wonder if it was a server glitch or a client issue - or a combination of both?

**Comment by b00ga** [comment]
Tried again today and seems to be working  so far.

**Comment by carlosefr** [comment]
It's working for me now as well.

**Comment by bookworm17** [comment]
I also have this problem:

iOS 16.3
iPad Pro 11in
Mastodon app v1.4.9
mastodon.social

**Comment by tyler-8** [comment]
This issue popped up for me today too.

IOS 16.1.2
App v1.4.9

**Comment by bookworm17** [comment]
Update: it was a content blocker. Apparently, they are a long ways behind the times. I filed a report with them to unblock for everyone.

**Comment by ocean90** [comment]
@bookworm17 Could you please explain what you mean by content blocker?

I also have this issue for several days now. Already tried a reinstall without luck.

**Comment by bookworm17** [comment]
Some DNS lookup services check the names against lists of malware and/or undesirable content names.

I had forgotten that one of my connection methods was using such a service. (The service was probably blocking everything in .social: this was reasonable a while ago, but needs more nuance now.) Once I configured the service to not block mastodon.social, it worked fine. I also notified the service that it should remove that block globally.

Craig

**Comment by carrythebanner** [comment]
I'm seeing this as well, app version 1.50 running on iOS 16.3.1, using an account on the mastodon.social instance. Using the web (Safari) on the same device, my home feed loads without issue. This first started occurring for me a couple of days ago.

**Comment by robertmryan** [comment]
* mastodon.social
* Mastodon for iOS v2023.8 (317)
* iPad running iOS 16.5

FWIW, this thread says “but everything else worked”. I would caveat that and say that not *everything* works. I.e., I can see old content (as of May 3rd), but I cannot see new content nor can I post anything. 

- - -

FWIW, I've uninstalled the app and reinstalled and it is working again.

**Comment by btb** [comment]
Same issue here. Latest app and iOS version. I’m signed in to two accounts, one at mastodon.social and one on another server (which also ends in ".social"). The mastodon.social account sometimes works but then regularly the “Home” tab will be stuck “Offline” for 24-48 hours at a time - “Posts” “Community” et al continue to work fine however. The account on the other server never has any trouble whatsoever.
I also have Ivory installed, and my mastodon.social timeline has no trouble loading there.

reinstalling the app has no effect

**Comment by bookworm17** [comment]
Apologies for the delay in replying.

In my case, I had a filter configured on my wireless network that I’d forgotten about. (It checked for possibly malicious sites.) I assume that it matched on the “.social” part of the name. I deactivated the filter and reported the false-positive to the vendor and it has worked fine[*] since.

[*] I mainly read the site on my iPad. Occasionally, it gets hung up on the connection. I just end the application and restart it and its fine.

Craig

**Comment by Mormontales** [comment]
Experiencing the same issue, but the profile feed isn't working as well as search or any other service. Uninstalled the app, but can't log back in, it returns HTTP503 error.

 ![image](https://github.com/mastodon/mastodon-ios/assets/138831467/40c62355-ebe8-4379-8037-d7d8a73b35d0)
![image](https://github.com/mastodon/mastodon-ios/assets/138831467/543023ee-4730-4325-a282-bfc3b37cd76f)

**Comment by j-f1** [comment]
@Mormontales per https://status.mastodon.social, mastodon.social is currently down, so that’s why you can’t connect.

**Comment by oaustegard** [comment]
Same problem, v2023.11, iOS 16.5.1

Will try reinstall

**Comment by oaustegard** [comment]
...after reinstall. Annoying ![image](https://github.com/mastodon/mastodon-ios/assets/3268739/9b3eba66-16bf-4dad-9b6f-b1d4f9906077)

**Comment by tpolecat** [comment]
Started having this issue yesterday. iOS 16.6, v2023.11

**Comment by bookworm17** [comment]
While my original issue has been closed (due to unanticipated URL filtering), I am running into a minor annoyance that may be relevant. I run the app (on an iPad, all software current versions) and it works. It works for 2-3 days, then shows "offline". If I close the app and start again, it works. The app may have problems suspending/reopening connections.

**Comment by wimsymons** [comment]
I have this issue as well. 

Even on the latest iOS version (v2023.15). 

My mastodon server is mastodon.social. 

Everything works for a few hours after I log out and back in again. 

But the next day, the Home does not update with new content, when you refresh the "offline" red pill appears. My avatar also disappears in the  account switcher in the bottom right corner. 

When I log out and back in again, everything works again. 

I don't need to enter credentials again in the browser window, but that is another (reported) bug. 

I don't know how the app handles authentication, but either some token refresh is no longer working or server sessions time out too soon.

**Comment by arietveld** [comment]
I have the same issue and not a solution. Also in de profile image circle there is no image. 

iPhone 11
IOS 17.2
mastodon for iPhone rel 2023.15
Server: mastodon.radio
![IMG_9315](https://github.com/mastodon/mastodon-ios/assets/23239773/87cdedf2-f08b-4a4a-b5d6-45523495db02)

**Comment by sallyFoster** [comment]
For me once I log in and out works fine. However, as soon as I close the window and reopen it, it shows as offline.

**Comment by squio** [comment]
Got the same on iPhone 13 mini, latest iOS 17 and most recent Mastodon app

Verified on four different wifi networks (one DNS filtered, the others open). 

Mastodon account on mastodon.online 

Interestingly the same Mastodon app on iPad Pro works just fine (only tested on DNS filtered WiFi)

Logging out and in again on iPhone does not work.

**Comment by andyblaha** [comment]
I experience similar:

since updating my iPad Air 2 (iOS 17.1.1) to Mastodon for iOS v2023.14 I got (and still get) the Offline message. So, as a trained user, I tried to log off and on again, but logging off crashes the app immediately and totally, after a new start I seem to be logged on still. Updating to 2023.15 did not help. Meanwhile, all is fine in the webbrowser (safari on the device), so I changed the password and hoped I'd be logged off by now or at least being asked for my password. Nada. 
I searched for closed issues, but cannot find a related and solved issue anywhere.
Strangely, I was using the app without the update on the iPhone in parallel which worked nicely, but during a switch of phones the app was updated to 2023.15 but still works, so maybe in my case it is an iPad issue? Any ideas?

**Comment by squio** [comment]
Quick confirmation this has to do with an extra account which has been deactivated (access token or similar)

Platform: iOS, everything up to date

My case:
- one mastodon.online account (active)
- one pixelfed account (got de-activated by expiring or something)

I forgot about the pixelfed account and it was not visible through the account switch button anymore when it worked on longer (!)

Finally logged in to pixelfed again which:
- left the state "offline" with all the symptoms from before
- now showed two (!) similar pixelfed accounts under the account switch

After removing both pixelfed accounts everything went back to normal (with just the mastodon.online account)

Bottom line: a deactivated account can become invisible but still very persistent (I even uninstalled the app, the account(s) were still there after re-installing from the App Store)

**Comment by arietveld** [comment]
Thanks squio! I also was using pixelfed. Your suggestion fixed my problem.

**Comment by ma4nn** [comment]
> Bottom line: a deactivated account can become invisible but still very persistent (I even uninstalled the app, the account(s) were still there after re-installing from the App Store)

Yep, same here, it was another account that became deactivated. Thanks very much!

**Comment by zeitschlag** [comment]
Thank you very much for the hint to pixelfed @squio. Using the Mastodon-app with a pixelfed-account would have never crossed my mind but after having created a pixelfed-account I can confirm: It doesn't work (although it might have worked in the past).

But this doesn't explain why several mastodon-servers seem to be affected, too, while being the only account.

**Comment by zeitschlag** [comment]
Another hint: While developing I ran into the [rate limits](https://docs.joinmastodon.org/api/rate-limits/):

> All endpoints and methods can be called 300 times within 5 minutes.

Getting a `429` from my instance all the time points out that I'm a heavy Mastodon-user. At least I see the "Offline"-banner regularly now.

**Comment by wildeng** [comment]
> Quick confirmation this has to do with an extra account which has been deactivated (access token or similar)
> 
> 
> 
> Platform: iOS, everything up to date
> 
> 
> 
> My case:
> 
> - one mastodon.online account (active)
> 
> - one pixelfed account (got de-activated by expiring or something)
> 
> 
> 
> I forgot about the pixelfed account and it was not visible through the account switch button anymore when it worked on longer (!)
> 
> 
> 
> Finally logged in to pixelfed again which:
> 
> - left the state "offline" with all the symptoms from before
> 
> - now showed two (!) similar pixelfed accounts under the account switch
> 
> 
> 
> After removing both pixelfed accounts everything went back to normal (with just the mastodon.online account)
> 
> 
> 
> Bottom line: a deactivated account can become invisible but still very persistent (I even uninstalled the app, the account(s) were still there after re-installing from the App Store)
> 
> 
> 
> 

Thanks a lot for this, after logging out from the Pixelfed account everything started working again!

**Comment by zaxbux** [comment]
I can also confirm. Once I removed a mastodon.lol account (I guess their server has shutdown), my other accounts started working. Mastodon for iOS 2023.16 (6043) on iPad Air 4th gen.

**Comment by tkimnguyen** [comment]
I just had to log out then back in again. I wasn't getting content updates for weeks now, no notifications. I also had posted a couple of times but the posts never actually made it out. Current versions: app 2023.6, iOS 17.2.1

**Comment by magnus919** [comment]
I hadn't seen any new posts on my Fosstodon account since Nov 2. After reading comments here, I logged out of my Pixelfed account and then the Fosstodon posts started flooding in.

**Comment by electricbrain-code** [comment]
Me too….
Mastodon for iOS v2024.11 (6903)
iOS 18.1.1 (22B91)
iPhone 14
MPVA3ZP/A
